src/main/java/randoop/reflection/OperationExtractor.java

```diff
@@ -38,55 +39,117 @@
   /** The collection of operations */
   private final Collection<TypedOperation> operations;
 
-  /** The class type of the declaring class for the collected operations */
+  /** The type of the declaring class for the collected operations */
   private ClassOrInterfaceType classType;
 
+  /** The predicate to test whether to omit an operation */
+  private OmitMethodsPredicate omitPredicate;
+
   /**
    * Creates a visitor object that collects the {@link TypedOperation} objects corresponding to
-   * members of the class type and satisfying the given predicate.
+   * members of the class satisfying the given visibility and reflection predicates and that don't
+   * violate the omit method predicate.
+   *
+   * <p>Once created this visitor should only be applied to members of {@code
+   * classType.getRuntimeType()}.
    *
-   * @param classType the declaring classtype for collected operations
-   * @param operations the collection of operations
+   * @param classType the declaring class for collected operations
+   * @param operations the collection of operations, will be side-effected
    * @param predicate the reflection predicate
+   * @param omitPredicate the list of {@code Pattern} objects for omitting methods, may be null
    * @param visibilityPredicate the predicate for test visibility
    */
   public OperationExtractor(
       ClassOrInterfaceType classType,
       Collection<TypedOperation> operations,
       ReflectionPredicate predicate,
+      OmitMethodsPredicate omitPredicate,
       VisibilityPredicate visibilityPredicate) {
     this.classType = classType;
     this.operations = operations;
     this.predicate = predicate;
     this.visibilityPredicate = visibilityPredicate;
+    this.omitPredicate = omitPredicate;
```

Comment:
Manipulate fields in the same order they are declared and the same order they are declared in the class.  (Otherwise it's too hard to keep track of them in your head.)  I suggest that `operations` should be last, because it is an output whereas all the others are inputs.  But, you might eliminate `operations` per my comment in email.